modules/tail-sampling-redis-store/src/main/scala/io/janstenpickle/trace4cats/sampling/tail/redis/RedisSampleDecisionStore.scala

```diff
@@ -90,11 +89,9 @@ object RedisSampleDecisionStore {
             Applicative[F].unit
           else
             for {
-              results <- decisions.foldLeft(Applicative[F].pure(List.empty[Fiber[F, Unit]])) {
-                case (acc, (traceId, decision)) =>
-                  cmd.setEx(keyPrefix -> traceId, decision, ttl).start.flatMap(fiber => acc.map(fiber :: _))
+              _ <- decisions.toList.parTraverse_ { case (traceId, decision) =>
```

Comment:
This is a much simpler implementation. `Parallel` for any IO performs applicative product via `both` - this is safer and prevents fiber leakage.

Reply:
It's worth backporting